modules/calendar/admin.php

```diff
@@ -4,15 +4,35 @@
 require_once("includes/admin_helper.php");
 
 // get bootstrap
-  call_required_libraries(array("jquery-min-3-1-1","bootstrap"));
-
+call_required_libraries(array("jquery-min-3-1-1", "bootstrap", "iziModalToast"));
 
 // if update category button is used
 if($_POST['updateCat'] == 1) {
-  // call for creation/updation of category
-  createUpdateCategory($_SESSION['category'], $_POST['catName'], $_POST['catCol'], 
-    $_POST['catDes'], $_POST['catType'], $_POST['catDur'], 
-    $_POST['catAllDay'], $_SESSION['category'] == "__NEW__");
+  // if this is a new category, check if there exists a category already with the same name
+  $nameCheck = false;
+  $categories = getCategories(); // retrieve up-to-date list of categories from admin_helper
+  if ($_SESSION['category'] == "__NEW__") {
+    foreach ($categories as $category) {
+      if ($category['pc_catname'] == $_POST['catName']) {
+        $nameCheck = true;
+        echo ('
+        <script type="text/javascript">
```

Comment:
This wont work call it at the bottom of the script so that it can affect the document
```
if(namecheck){
echo
}
```

Reply:
You might want to alert this at the bottom of the script rather see https://github.com/LibreHealthIO/lh-ehr/blob/master/interface/usergroup/facilities_add.php It should help